### PR TITLE
[mlir][nvgpu][nvvm] Add myself as a primary reviewer for nvgpu and nvvm dialects

### PR DESCRIPTION
**Pull request by grypp**

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -106,5 +106,16 @@ clang/test/AST/Interp/ @tbaederr
 # MLIR Sparsifier.
 /mlir/**/*SparseTensor*/ @aartbik @PeimingLiu @yinying-lisa-li @matthias-springer
 
+# MLIR NVGPU Dialect
+/mlir/**/NVGPU/ @grypp
+/mlir/**/NVGPUTo*/ @grypp
+/mlir/**/NVGPUTo*/ @grypp
```
Comment (joker-eph) [inline .github/CODEOWNERS:112]: What's the difference between the last two lines?

Also can you compress the three lines into: `mlir/**/NVGPU*/` ?

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -106,5 +106,16 @@ clang/test/AST/Interp/ @tbaederr
...
+/mlir/**/sm90*/ @grypp
```
Comment (joker-eph) [inline .github/CODEOWNERS:113]: Is this catching the integration tests? don't you want to match something like `/mlir/test/**/CUDA*/` ?

File: .github/CODEOWNERS [hunk 3]
```diff
@@ -106,5 +106,16 @@ clang/test/AST/Interp/ @tbaederr
...
+
+# MLIR NVVM Dialect in MLIR
+/mlir/**/LLVMIR/BasicPtxBuilderInterface* @grypp
+/mlir/**/LLVMIR/NVVM* @grypp
+/mlir/**/NVVMToLLVM*/ @grypp
```
Comment (joker-eph) [inline .github/CODEOWNERS:118]: That won't catch `mlir/include/mlir/Target/LLVM/NVVM`
Why not just do `mlir/**/NVVM*/`?

File: .github/CODEOWNERS (hunk 2 above)
Comment (grypp) [reply .github/CODEOWNERS:113]: I could catch actually all of the integration tests,

**Review by joker-eph — APPROVED** [review]

**Pull request merged**